src/Components/Buttons/CatalogButtons.tsx

```diff
@@ -121,16 +126,19 @@ class CatalogButtons extends Component<Props,ICatalogButtons>{
      * @param btn
      * @returns {string|string|*}
      */
-    getVersion(btn) {
+    // TODO: @Tim: Could this be solved in a more dynamic way, f.e. storing currentICD/DRG/CHOP/TARMED in a hash and
```

Comment:
Yes this should work with hashes. If it does not work directly with state variables you should be able do to: this.state[hash[btn]]
Javascript attributes of an object can always be accessed by string hash access as well.